src/swish/db.ss

```diff
@@ -164,20 +164,22 @@
                [worker #f]))))
 
   (define (terminate reason state)
-    (catch (flush state))
-    (vector-for-each (lambda (e) (sqlite:finalize (entry-stmt e)))
-      (hashtable-values (cache-ht ($state cache))))
-    (finalize-lazy-statements ($state cache))
-    (sqlite:close ($state db))
-    'ok)
+    (let ([x (catch (flush state))])
+      (vector-for-each (lambda (e) (sqlite:finalize (entry-stmt e)))
+        (hashtable-values (cache-ht ($state cache))))
+      (finalize-lazy-statements ($state cache))
+      (sqlite:close ($state db))
+      (match x
+        [#(EXIT ,reason) (raise reason)]
+        [,_ 'ok])))
```

Comment:
Ah, right. My WIP commit message was really unclear. I had left it as `catch` and `#(EXIT ...)` with the intent of squashing it into the "fix: bad input to db:log could hang db:stop indefinitely" commit. I've added the missing commit that flips these to the new style to be squashed in as part of the "retain more debugging context" commit. Hopefully this will all make sense in the final rebase.